packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -82,9 +81,23 @@ export const MenuButton = compose<MenuButtonType>({
         <Slots.button />
         {context.showContextualMenu && (
           <Slots.contextualMenu>
-            {menuItems.map((menuItem) => (
-              <ContextualMenuItem key={menuItem.itemKey} {...menuItem} />
-            ))}
+            {menuItems.map((menuItem) => {
+              const { hasSubmenu, submenu, showSubmenu, componentRef, submenuItems, ...items } = menuItem;
+              return hasSubmenu ? (
+                <Slots.contextualMenuItems>
+                  <SubmenuItem componentRef={componentRef} {...items} />
+                  {showSubmenu && (
+                    <Submenu target={componentRef} {...submenu}>
+                      {submenuItems &&
+                        submenuItems.map &&
+                        submenuItems.map((submenuItem) => <ContextualMenuItem key={submenuItem.itemKey} {...submenuItem} />)}
```

Comment:
Is the check for `submenuItems.map` necessary? If the `submenuItems` array is defined, calling `submenu.map` should always work, right? Even if the array is empty?

nit: I also think you could simplify this by writing `submenuItems?.map((....))` but checking if submenuItems is defined prior to calling map is fine too

Reply:
Yeah, this option is better, thanks `submenuItems?.map((....))`